ir/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -40,12 +42,29 @@ object Serializers {
     new Serializer().serialize(stream, classDef)
   }
 
-  def deserializeEntryPointsInfo(stream: InputStream): EntryPointsInfo = {
-    new Deserializer(stream).deserializeEntryPointsInfo()
-  }
+  /** Deserializes entry points from the given buffer.
+   *
+   *  @throws java.nio.BufferUnderflowException if not enough data is available
+   *      in the buffer. In this case the buffer's position is unspecified and
+   *      needs to be reset by the caller.
+   */
+  def deserializeEntryPointsInfo(buf: ByteBuffer): EntryPointsInfo =
+    withBigEndian(buf)(new Deserializer(_).deserializeEntryPointsInfo())
 
-  def deserialize(stream: InputStream): ClassDef = {
-    new Deserializer(stream).deserialize()
+  /** Deserializes a class def from the given buffer.
+   *
+   *  @throws java.nio.BufferUnderflowException if not enough data is available
+   *      in the buffer. In this case the buffer's position is unspecified and
+   *      needs to be reset by the caller.
+   */
+  def deserialize(buf: ByteBuffer): ClassDef =
+    withBigEndian(buf)(new Deserializer(_).deserialize())
+
+  @inline
+  private def withBigEndian[T](buf: ByteBuffer)(body: ByteBuffer => T): T = {
+    val o = buf.order()
+    try body(buf)
```

Comment:
Didn't you forget to actually call `buf.order(ByteOrder.BIG_ENDIAN)` here?

Reply:
Yes...